### PR TITLE
Filter report structure based on filters

### PR DESCRIPTION
**Pull request by AdityaKhatri**
## Changes

* Filter report structure based on filters

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: app/views/Project/Tagging/Export/utils.ts [hunk 1]
```diff
@@ -131,32 +132,169 @@ function transformLevelsColumnFirst(
...
+            (val) => {
+                // NOTE: Levels are by default a composite of keys
+                // We are checking for any item in the list of keys to include all children
+                // of selected root as well
+                // Sample level: {rowId}-{subRowId}
+                // If row is selected in filter, we show all child of that row
+                const splits = level.id.split('-');
+                return splits.some((s) => s === val);
+            },
+        );
```
Comment (tnagorra) [inline app/views/Project/Tagging/Export/utils.ts]: We should do all of the pre-processing for valueList outside.

File: app/views/Project/Tagging/Export/utils.ts [hunk 2]
```diff
@@ -131,32 +132,169 @@ function transformLevelsColumnFirst(
...
+
+            // NOTE: If the node has children after being filtered
+            // or the node is selected itself, we show the node
+            if (nodes.length > 0 || isActiveKey) {
+                return ({
+                    key: level.id,
+                    title: level.title,
+                    selected: true,
+                    draggable: true,
+                    nodes: mapReportLevelsToNodes(level.sublevels, valueList),
```
Comment (tnagorra) [inline app/views/Project/Tagging/Export/utils.ts]: Let's just use nodes

File: app/views/Project/Tagging/Export/utils.ts [hunk 3]
```diff
@@ -131,32 +132,169 @@ function transformLevelsColumnFirst(
...
+                });
+            }
+        }
+
+        if (!level.sublevels && isActiveKey) {
```
Comment (tnagorra) [inline app/views/Project/Tagging/Export/utils.ts]:
```suggestion
        } else if (isActiveKey) {
```

File: app/views/Project/Tagging/Export/utils.ts [hunk 4]
```diff
@@ -131,32 +132,169 @@ function transformLevelsColumnFirst(
...
+            });
+        }
+
+        return undefined;
+    }).filter(isDefined);
+}
+
+// NOTE: This function is used to generate report levels for matrix2d
+// It currently filters out any item that is not part of the item in the 2 filter lists of matrix2d
+function mapReportLevelsToNodesForMatrix2d(
```
Comment (tnagorra) [inline app/views/Project/Tagging/Export/utils.ts]: Let's try to reuse this logic if we can

**Review by tnagorra — DISMISSED** [review]

**Review by samshara — DISMISSED** [review]

**Review by samshara — APPROVED** [review]

**Pull request merged**